### ISSUE TITLE
GraphicsView not shown / issue with IsVisible property

### ISSUE THREAD

**Issue by espenrl** [dotnet/maui#12903]
### Description

When IsVisible has been set to false and then back to true the GraphicsView is not shown. In fact the drawable is not ever called.

### Steps to Reproduce

Create a `GraphicsView` with `IsVisible = false` initially.

### Link to public reproduction project repository

N/A

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 12

### Did you find any workaround?

``` CSharp
GraphicsViewHandler.Mapper.AppendToMapping(nameof(IGraphicsView.Visibility), MapVisibility);

private static void MapVisibility(IGraphicsViewHandler handler, IGraphicsView graphicsView)
{
    handler.PlatformView.Visibility = graphicsView.Visibility.ToPlatformVisibility();
}
```

**Comment by jsuarezruiz** [comment]
Could you attach a sample where reproduce the issue?. Cannot reproduce it.
![issue-12903](https://user-images.githubusercontent.com/6755973/215047841-8e594102-2793-4f06-8a90-8081dbb32cc2.gif)

**Comment by espenrl** [comment]
https://github.com/espenrl/issue-repro/blob/main/maui-android-graphicsview

Finally had the time to make a sample. I found out that the combination of `IsVisible="False"` and `InputTransparent="True"` is what triggers the issue.

**Comment by mattleibow** [comment]
I found the issue and I think this is all to do with the fact that when we do things that change the ContainerView, we are not correctly setting and unsetting the properties.

**Comment by espenrl** [comment]
Yes, the dynamic creation and destruction of ContainerView depending upon what properties are set makes it hard to track what properties to "transfer" between PlatformView and ContainerView. Might as weel always have the ContainerView for the lifetime of a view. Would that be an idea?

**Comment by espenrl** [comment]
IElementHandler.SetVirtualView() initializes a view with its property mappers. Some mappers run before creating a ContainerView, some run after. A few propery mappers that took action on the PlatformView now take action on the ContainerView. That leaves the view in an invalid state for the remainder of the lifetime. This is the cause of many of my problems regarding Opacity and IsVisible for all views, not just GraphicsView.

**Comment by espenrl** [comment]
### These four trigger the ContainerView property mapper

https://github.com/dotnet/maui/blob/04bfb275d9401a631d3e9a1e20665e68e99b79c4/src/Core/src/Handlers/View/ViewHandler.cs#L246-L258

https://github.com/dotnet/maui/blob/04bfb275d9401a631d3e9a1e20665e68e99b79c4/src/Core/src/Handlers/View/ViewHandler.cs#L291-L296

https://github.com/dotnet/maui/blob/04bfb275d9401a631d3e9a1e20665e68e99b79c4/src/Core/src/Handlers/View/ViewHandler.cs#L326-L334

### ContainerView property mapper

https://github.com/dotnet/maui/blob/04bfb275d9401a631d3e9a1e20665e68e99b79c4/src/Core/src/Handlers/View/ViewHandler.cs#L273-L281

### ContainerView infrastructure (Android)

https://github.com/dotnet/maui/blob/04bfb275d9401a631d3e9a1e20665e68e99b79c4/src/Core/src/Handlers/View/ViewHandler.cs#L81-L96

https://github.com/dotnet/maui/blob/04bfb275d9401a631d3e9a1e20665e68e99b79c4/src/Core/src/Handlers/View/ViewHandler.cs#L98-L101

https://github.com/dotnet/maui/blob/04bfb275d9401a631d3e9a1e20665e68e99b79c4/src/Core/src/ViewExtensions.cs#L51-L73

https://github.com/dotnet/maui/blob/04bfb275d9401a631d3e9a1e20665e68e99b79c4/src/Core/src/Handlers/View/ViewHandlerOfT.Android.cs#L20-L37

**Comment by espenrl** [comment]
What I can't figure out is why this problem manifest itself on Android, but not on iOS and Windows. And it seems logical that i should. It's probably something I don't know about property mapper initialization across the platforms.

**Comment by arahmancsd** [comment]
I am experiencing the same thing inside a `CollectionView` `ItemTemplate`.  

Place a GraphicsView item inside a data template in `CollectionView` with `IsVisible = {Binding ViewModelVisibleProperty}`. Then update the ViewModel accordingly with ViewModelVisibleProperty = True for the specific CollectionView item. The GraphicsView doesn't show up in the view.

**Comment by beeradmoore** [comment]
@jsuarezruiz I think this issue is the same as something I've just come across. If not please let me know and I'll file as a new issue.

Repro: https://github.com/beeradmoore/maui-issue-IsVisibleBindingTest

This repro project is using CommunityToolkit.MVVM. I have a `MainPageModel` which has a property called`IsSelected`.  `IsSelected` is false by default.

There is a red BoxView in the middle of the page which has its `IsVisible` bound to the `IsSelected` on the `MainPageModel`. So its `IsVisible` is false by default.

`IsSelected` is toggled by a button in the top left that says "Toggle IsSelected: {current state true/false}".

When I run this in the iOS simulator I have no red box (as expected) but when I hit the button I still do not have a red box. On macOS I have the same behaviour. 

HOWEVER, on macOS if I take focus away from the window the red box will appear.
On iOS rotating the simulator does not make it appear.

In the project (not the repro) I have that I found this problem with if I resize the window on macOS the box will appear. In that project the box is a SkiaSharp SKCanvasView.  I also observe there that its X/Y/Width/Height gets measured for the first time when the window is resized. This me think its size/layout is not calcualted when it is added to the window.

I added a `BoxView_OnPropertyChanged` event onto the BoxView but I don't think it's doing anything odd.

```
BoxView_OnPropertyChanged - HorizontalOptions
BoxView_OnPropertyChanged - VerticalOptions
BoxView_OnPropertyChanged - WidthRequest
BoxView_OnPropertyChanged - HeightRequest
BoxView_OnPropertyChanged - Parent
BoxView_OnPropertyChanged - IsVisible
BoxView_OnPropertyChanged - Window
-> clicked button here
BoxView_OnPropertyChanged - IsVisible
BoxView_OnPropertyChanged - X - 749
BoxView_OnPropertyChanged - Y - 381
BoxView_OnPropertyChanged - Width - 100
BoxView_OnPropertyChanged - Height - 100
```

I initially thought it was not measuring its x/y/w/h and thats why it wasn't visible. But this shows it does measure it, just not when the page is laid out.


I tested all lof this on an Android Emulator and it appeared to perform correctly.

**Comment by beeradmoore** [comment]
@kubaflo , I have updated the repo in my comment above and the sample now works. There is no red box, I click the toggle button and the red box appears. This is different to the behaviour I was experiencing in 2024.

**Comment by kubaflo** [comment]
Thanks @beeradmoore — the updated repro now behaves correctly (no red box initially, then it appears on toggle), which is a clear change from the 2024 behaviour you originally hit.

Closing as fixed. @espenrl — if this still reproduces for you on .NET 10, comment here and it will be reopened.

> Disclosure: this comment was generated by AI.
